### PR TITLE
[ADF-994] Integrate SFS enhancements for Pagination component

### PR DESCRIPTION
**Pull request by chris-equis**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [x] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: ng2-components/ng2-alfresco-core/src/components/pagination/pagination.component.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<div class="adf-pagination__block adf-pagination__range">
+    <span>{{ 'APP.PAGINATION.SHOW_RANGE' | translate: { range: range.join('-'), total: totalItems } }}</span>
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-core/src/components/pagination/pagination.component.html]: Translation keys should be present in the `src/i18n/en.json` and probably needs "CORE" namespace for that as that file is currently empty.

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: ng2-components/ng2-alfresco-core/src/components/pagination/pagination.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# Pagination Component
+
+## Basic example
+
+```html
+<adf-pagination
+    [pagination]="pagination"
+    (onPaginationEvent)="doSomething($event)"
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-core/src/components/pagination/pagination.md]: I think those events are not implemented or replaced by the 'change' event.

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: ng2-components/ng2-alfresco-core/src/components/pagination/pagination.component.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    host: { 'class': 'adf-pagination' },
+    templateUrl: './pagination.component.html',
+    styleUrls: [ './pagination.component.scss' ],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None
+})
+export class PaginationComponent {
+
+    @Input()
+    pageSizes: number[] = [ 25, 50, 100 ];
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-core/src/components/pagination/pagination.component.ts]: Needs to be "supportedPageSizes " to provide backwards compatibility.

File: ng2-components/ng2-alfresco-core/src/components/pagination/pagination.component.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+    @Input()
+    pagination: Pagination = null;
+
+    @Output()
+    change: EventEmitter<PaginationQueryParams> = new EventEmitter<PaginationQueryParams>();
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-core/src/components/pagination/pagination.component.ts]: Needs the following events emitted to provide backwards compatibility
```ts
@Output()
changePageSize: EventEmitter<Pagination> = new EventEmitter<Pagination>();

@Output()
nextPage: EventEmitter<Pagination> = new EventEmitter<Pagination>();

@Output()
prevPage: EventEmitter<Pagination> = new EventEmitter<Pagination>();
```

File: ng2-components/ng2-alfresco-datatable/src/components/pagination/pagination.component.ts [hunk 5]
```diff
@@ -20,7 +20,7 @@ import { Pagination } from 'alfresco-js-api';
 import { PaginationData } from '../../models/pagination.data';
 
 @Component({
-    selector: 'adf-pagination, alfresco-pagination',
+    selector: 'alfresco-pagination',
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-datatable/src/components/pagination/pagination.component.ts]: Move both selectors to the core/pagination component and remove this implementation once backwards compatibility features are introduced.

File: ng2-components/ng2-alfresco-documentlist/src/components/document-list.component.html [hunk 6]
```diff
@@ -47,10 +47,11 @@
         </loading-content-template>
     </div>
 </adf-datatable>
-<adf-pagination *ngIf="isPaginationEnabled()"
+
+<alfresco-pagination *ngIf="isPaginationEnabled()"
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-documentlist/src/components/document-list.component.html]: Use adf-pagination here once the old one is deprecated

File: ng2-components/ng2-alfresco-search/src/components/search.component.html [hunk 7]
```diff
@@ -68,13 +68,13 @@
                 </content-action>
             </content-actions>
         </adf-document-list>
-        <adf-pagination
+        <alfresco-pagination
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-search/src/components/search.component.html]: Use adf-pagination here once the old one is deprecated

**Review by DenysVuika — CHANGES_REQUESTED** [review]
-

File: ng2-components/ng2-alfresco-core/src/components/pagination/pagination.md (hunk 2 above)
Comment (chris-equis) [inline ng2-components/ng2-alfresco-core/src/components/pagination/pagination.md]: Will update documentation soon

**Comment by eromano** [comment]
I think is missing a link to the pagination.md in the main Readme of the document. If we don't add it will be difficult for people find it

**Comment by eromano** [comment]
there are ✖ 13 tests failed need to be fixed before to merge this story

**Comment by chris-equis** [comment]
@eromano Travis passes, is there anything else for this PR? Thanks!

**Pull request merged**